store/sqlstore/channel_store.go

```diff
@@ -2976,73 +2995,71 @@ func (s SqlChannelStore) autocompleteInTeamForSearchDirectMessages(userId string
 	return channels, nil
 }
 
-// TODO: rewrite in squirrel (https://github.com/mattermost/mattermost-server/issues/19333)
 func (s SqlChannelStore) SearchInTeam(teamId string, term string, includeDeleted bool) (model.ChannelList, error) {
-	deleteFilter := "AND c.DeleteAt = 0"
-	if includeDeleted {
-		deleteFilter = ""
+
+	where := sq.And{sq.Eq{"c.TeamId": teamId}}
+	if !includeDeleted {
+		where = append(where, sq.Eq{"c.DeleteAt": 0})
 	}
 
-	return s.performSearch(`
-		SELECT
-			Channels.*
-		FROM
-			Channels
-		JOIN
-			PublicChannels c ON (c.Id = Channels.Id)
-		WHERE
-			c.TeamId = :TeamId
-			`+deleteFilter+`
-			SEARCH_CLAUSE
-		ORDER BY c.DisplayName
-		LIMIT 100
-		`, term, map[string]interface{}{
-		"TeamId": teamId,
-	})
+	if term != "" {
+		searchClause, err := s.searchClause(term)
+		if err != nil {
+			return nil, errors.Wrapf(err, "SearchInTeam_searchClause")
+		}
+		where = append(where, searchClause)
+	}
+
+	query := sq.Select("Channels.*").
+		From("Channels").
+		Join("PublicChannels c ON (c.Id = Channels.Id)").
+		Where(where).
+		OrderBy("c.DisplayName").
+		Limit(100)
+
+	result, err := s.performSearch(query, term)
+	if result == nil {
+		return model.ChannelList{}, err
+	}
+	return result, err
 }
 
-// TODO: rewrite in squirrel (https://github.com/mattermost/mattermost-server/issues/19334)
 func (s SqlChannelStore) SearchArchivedInTeam(teamId string, term string, userId string) (model.ChannelList, error) {
-	publicChannels, publicErr := s.performSearch(`
-		SELECT
-			Channels.*
-		FROM
-			Channels
-		JOIN
-			Channels c ON (c.Id = Channels.Id)
-		WHERE
-			c.TeamId = :TeamId
-			SEARCH_CLAUSE
-			AND c.DeleteAt != 0
-			AND c.Type != :ChannelType
-		ORDER BY c.DisplayName
-		LIMIT 100
-		`, term, map[string]interface{}{
-		"TeamId":      teamId,
-		"UserId":      userId,
-		"ChannelType": model.ChannelTypePrivate,
-	})
 
-	privateChannels, privateErr := s.performSearch(`
-		SELECT
-			Channels.*
-		FROM
-			Channels
-		JOIN
-			Channels c ON (c.Id = Channels.Id)
-		WHERE
-			c.TeamId = :TeamId
-			SEARCH_CLAUSE
-			AND c.DeleteAt != 0
-			AND c.Type = :ChannelType
-			AND c.Id IN (SELECT ChannelId FROM ChannelMembers WHERE UserId = :UserId)
-		ORDER BY c.DisplayName
-		LIMIT 100
-		`, term, map[string]interface{}{
-		"TeamId":      teamId,
-		"UserId":      userId,
-		"ChannelType": model.ChannelTypePrivate,
-	})
+	baseWhere := sq.And{
```

Comment:
Same as earlier. Let's use the new style. And inline some of the sub-queries for better readability. And use model.ChannelTypePrivate instead of "P"